shift.go

```diff
@@ -1,10 +1,39 @@
 // Package shift provides the persistence layer for a simple "finite state machine"
 // domain model with validation, explicit fields and reflex events per state change.
 //
+// Defining an FSM
+//
 // shift.NewFSM builds a FSM instance that allows specific mutations of
 // the domain model in the underlying sql table via inserts and updates.
 // All mutations update the status of the model, mutates some fields and
 // inserts a reflex event.
+//
+// There needs to be at least one insert status defined and 0..n update
```

Comment:
Suggestion:

A shift state machine is defined by multiple states. States have the following properties:
 - A state has a name; defined as a Status enum instance.
 - A state has an associated data struct defining the data required to enter the state.
 - A state has an allowed set of next states.
 - Only one state can be the initial state.
 - All other states are reached by explicit transitions from the initial state.
 - It is allowed to transition to a previous state or even to itself.
 - The states and transitions form a cyclical directed graph with a single root node.